### PR TITLE
Fix autoreload with embedded server

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-2214](https://youtrack.jetbrains.com/issue/KTOR-2214)

### CODE REVIEW

File: ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/TestServer.kt [hunk 1]
```diff
@@ -22,21 +22,28 @@ internal fun startServer(): Closeable {
...
+        Thread.sleep(1000)
+    } catch (cause: Throwable) {
+        scope.close()
+    }
 
     return Closeable {
-        tcpServer.close()
-        server.stop(0L, 0L, TimeUnit.MILLISECONDS)
-        tlsServer.stop(0L, 0L, TimeUnit.MILLISECONDS)
+        scope.close()
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/TestServer.kt:46]: can you return `scope` directly?

File: ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/CloseableScope.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.client.tests.utils.tests
+
+import java.io.*
+
+class CloseableScope : Closeable {
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/CloseableScope.kt:9]: Minor: `Scope` in the name makes me think about `CoroutineScope`. Can you think of some other name?

**Review by rsinukov — APPROVED** [review]

File: ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/TestServer.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/TestServer.kt:46]: Nice catch, thanks

File: ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/CloseableScope.kt (hunk 2 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/CloseableScope.kt:9]: Do you have any advice?

Comment (rsinukov) [reply ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/tests/CloseableScope.kt:9]: `ClosableGroup` or something similar? But it's fine if you leave it like that

**Pull request merged**